File: Cal_Distance/cal_distance.js
$(function(){
   $("#get_coordinate").click(function(){
    var city_A = $('#city_A').val();
    var city_B = $('#city_B').val();
    var url = 'https://restapi.amap.com/v3/geocode/geo';
    $.get(
        url,
        {
            key: "5c5cf68cefc1682dcbfdf84a3a7d88c2",
            address: city_A + '|' + city_B,
            batch:true
        }, function(data){
            $("#init_result1").html(data.geocodes[0].location) 
            $("#init_result2").html(data.geocodes[1].location)
        });
   });

   $("#cal_distance").click(function(){
       var st = $("#init_result1").html();
       var ed = $("#init_result2").html();
       var url2 = "https://restapi.amap.com/v3/distance";
       $.get(
           url2,
           {
            key:"33f11904550840bc85c725842a57bd04",   
            origins:st,
            destination:ed,
           },function(data){
               distance_meter = data.results[0].distance;
               $("#distance").html((distance_meter/1000).toFixed(1));
           });

   });

   $("#clear").click(function(){
       $("#city_A").val('');
       $("#city_B").val('');
       $("#init_result1").html('');
       $("#init_result2").html('');
       $("#distance").html('');
   });
})

File: Cal_Distance/Cal_Distance.py
import pandas as pd
import numpy as np
import requests




def main(name,column):
    data_Location = []
    data = pd.read_csv(name+'.csv',sep=',')
    Location = data[column].tolist()
    n = 10
    split_Location = [Location[i:i+n] for i in range(0,len(Location),n)]
    for item in split_Location:
        name_Location = '|'.join(item)
        data_Location.append(name_Location)
    get_coordinate(data_Location,data,name)

def get_coordinate(data_Location,data,name):
    Location_list = []
    for address in data_Location:
        url = 'https://restapi.amap.com/v3/geocode/geo';
        parameters = {'key': "5c5cf68cefc1682dcbfdf84a3a7d88c2",
                    'address': address,
                    'batch': True}
        response = requests.get(url,parameters)
        response_json = response.json()
        coordinate = response_json['geocodes']
        for item in coordinate:
            Location_list.append(item['location'])
    print(Location_list)
    data['Cood'] = Location_list
    data.to_csv('Update_'+ name +'.csv',encoding = 'utf_8_sig')

def cal_distance():
    print('cal_distance')

if __name__ == '__main__':
    name = 'Location'#'甘肃发货信息'
    column = 'Location'#'place_of_loading'
    main(name,column)